api/src/services/order.js

```diff
@@ -105,11 +106,13 @@ const requiredRequesterOrAnyClause = getRequiredAddressOrAnyClause('requester');
 const requiredBeneficiaryOrAnyClause =
   getRequiredAddressOrAnyClause('beneficiary');
 
-const getAddressOrAnyRestrictClause = (key) => (value) => {
+const getAddressOrAnyRestrictClause = (key) => (value, isStrict) => {
   const restrictKey = `order.${key}`;
   return (
     !isAny(value) && {
-      [restrictKey]: value ? { $in: [NULL_ADDRESS, value] } : NULL_ADDRESS,
+      [restrictKey]: value
+        ? { $in: isStrict ? [value] : [NULL_ADDRESS, value] }
+        : NULL_ADDRESS,
```

Comment:
we can do a little optimization for the DB here by not using `$in` if not needed

```suggestion
      [restrictKey]: value
        ? isStrict
          ? value
          : { $in: [NULL_ADDRESS, value] }
        : NULL_ADDRESS,
```

Reply:
je propose de supprimer les `[expressions](https://eslint.org/docs/latest/rules/no-nested-ternary)` ternaires imbriquées :


const getAddressOrAnyRestrictClause = (key) => (value, isStrict) => {
  const restrictKey = `order.${key}`;
  if (isAny(value)) {
    return null;
  }
  const result = { [restrictKey]: NULL_ADDRESS };
  if (value) {
    result[restrictKey] = isStrict ? value : { $in: [NULL_ADDRESS, value] };
  }
  return result;
};